lib/prometheus-metrics.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+const client = require('prom-client');
+
+// The current tests has circuit names like:
+// 'circuit one' (with blank space) and others like
+// 3beb8f49-62c0-46e0-b458-dcd4a62d0f48.
+// So to avoid "Error: Invalid metric name" we are changing the
+// circuit name to pass the tests.
+// More details:
+// https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels
+function normalizePrefix (prefixName) {
+  prefixName = prefixName.replace(/ /g, '_');
+  prefixName = prefixName.replace(/-/g, '_');
+  if (/^\d+$/.test(prefixName.charAt(0))) {
+    prefixName = `a${prefixName.substring(1)}`;
+  }
+  return `${prefixName}_`;
+}
+
+class PrometheusMetrics {
+  constructor (circuit) {
+    this.circuit = circuit;
+    this._client = client;
+    this.counters = [];
+    const prefix = normalizePrefix(`${this.circuit.name}_`);
```

Comment:
Seems like you don't need the `_` at the end here, do you? We're already adding it in the `normalizePrefix()` function.

Reply:
I keep that only to solve the issue with the tests.